Mathlib/Topology/ContinuousFunction/Basic.lean

```diff
@@ -325,12 +325,35 @@ def prodSwap : C(α × β, β × α) := .prodMk .snd .fst
 
 end Prod
 
+section Sigma
+
+variable {I A : Type*} {X : I → Type*} [TopologicalSpace A] [∀ i, TopologicalSpace (X i)]
+
 /-- `Sigma.mk i` as a bundled continuous map. -/
 @[simps apply]
-def sigmaMk {ι : Type*} {Y : ι → Type*} [∀ i, TopologicalSpace (Y i)] (i : ι) :
-    C(Y i, Σ i, Y i) where
+def sigmaMk (i : I) : C(X i, Σ i, X i) where
   toFun := Sigma.mk i
 
+/--
+To give a continuous map out of a disjoint union, it suffices to give a continuous map out of
+each term
+-/
+def sigma (f : ∀ i, C(X i, A)) : C((Σ i, X i), A) where
+  toFun ig := f ig.fst ig.snd
```

Comment:
I think this is
```suggestion
/--
To give a continuous map out of a disjoint union, it suffices to give a continuous map out of
each term
-/
def sigma (f : ∀ i, C(X i, A)) : C((Σ i, X i), A) where
  toFun := Sigma.curry fun i x => f i x
```
which is probably worth mentioning in the docstring, if you don't want this implementation.